ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ranger/RangerRestClientImpl.java

```diff
@@ -349,6 +354,90 @@ public boolean checkConnection(String url) {
+
+      resourceNameList.add(sourceDb);
+      rangerPolicyResource.setValues(resourceNameList);
+      RangerPolicy.RangerPolicyResource rangerPolicyResourceColumn =new RangerPolicy.RangerPolicyResource();
+      rangerPolicyResourceColumn.setValues(new ArrayList<String>(){{add("*"); }});
+      RangerPolicy.RangerPolicyResource rangerPolicyResourceTable =new RangerPolicy.RangerPolicyResource();
+      rangerPolicyResourceTable.setValues(new ArrayList<String>(){{add("*"); }});
+      rangerPolicyResourceMap.put("database", rangerPolicyResource);
+      rangerPolicyResourceMap.put("table", rangerPolicyResourceTable);
+      rangerPolicyResourceMap.put("column", rangerPolicyResourceColumn);
+      denyRangerPolicy.setResources(rangerPolicyResourceMap);
+
+      denyPolicyItemAccesses.add(new RangerPolicy.RangerPolicyItemAccess("create", true));
+      denyPolicyItemAccesses.add(new RangerPolicy.RangerPolicyItemAccess("update", true));
+      denyPolicyItemAccesses.add(new RangerPolicy.RangerPolicyItemAccess("drop", true));
+      denyPolicyItemAccesses.add(new RangerPolicy.RangerPolicyItemAccess("alter", true));
+      denyPolicyItemAccesses.add(new RangerPolicy.RangerPolicyItemAccess("index", true));
+      denyPolicyItemAccesses.add(new RangerPolicy.RangerPolicyItemAccess("lock", true));
+      denyPolicyItemAccesses.add(new RangerPolicy.RangerPolicyItemAccess("write", true));
+      denyPolicyItemAccesses.add(new RangerPolicy.RangerPolicyItemAccess("ReplAdmin", true));
+      denyPolicyItem.setAccesses(denyPolicyItemAccesses);
+      denyPolicyItemsForPublicGroup.add(denyPolicyItem);
+      List<String> denyPolicyItemsGroups = new ArrayList<String>();
+      denyPolicyItemsGroups.add("public");
+      denyPolicyItem.setGroups(denyPolicyItemsGroups);
+      denyRangerPolicy.setDenyPolicyItems(denyPolicyItemsForPublicGroup);
+
+      denyExceptionsPolicyItemAccesses.add(new RangerPolicy.RangerPolicyItemAccess("create", true));
+      denyExceptionsPolicyItemAccesses.add(new RangerPolicy.RangerPolicyItemAccess("update", true));
+      denyExceptionsPolicyItemAccesses.add(new RangerPolicy.RangerPolicyItemAccess("drop", true));
+      denyExceptionsPolicyItemAccesses.add(new RangerPolicy.RangerPolicyItemAccess("alter", true));
+      denyExceptionsPolicyItemAccesses.add(new RangerPolicy.RangerPolicyItemAccess("index", true));
+      denyExceptionsPolicyItemAccesses.add(new RangerPolicy.RangerPolicyItemAccess("lock", true));
+      denyExceptionsPolicyItemAccesses.add(new RangerPolicy.RangerPolicyItemAccess("write", true));
+      denyExceptionsPolicyItemAccesses.add(new RangerPolicy.RangerPolicyItemAccess("select", true));
+      denyExceptionsPolicyItemAccesses.add(new RangerPolicy.RangerPolicyItemAccess("read", true));
+      denyExceptionsPolicyItemAccesses.add(new RangerPolicy.RangerPolicyItemAccess("ReplAdmin", true));
```

Comment:
Aren't we handling ReplAdmin in another patch?

Reply:
No we are denying repladmin for all users except hive. In the other patch it will be disabled for hive as well